geode-assembly/src/acceptanceTest/java/org/apache/geode/metrics/GatewayReceiverMetricsTest.java

```diff
@@ -0,0 +1,307 @@
+        "--server-port=" + receiverServerPort,
+        "--classpath=" + metricsPublishingServiceJarPath,
+        "--J=-Dgemfire.distributed-system-id=" + receiverSystemId);
+
+    gfshRule.execute(startSenderLocatorCommand, startReceiverLocatorCommand,
+        startSenderServerCommand, startReceiverServerCommand);
+
+    String gatewaySenderId = "gs";
+
+    String connectToSenderLocatorCommand = "connect --locator=localhost[" + senderLocatorPort + "]";
+
+    String startGatewaySenderCommand = String.join(GFSH_COMMAND_SEPARATOR,
+        "create gateway-sender",
+        "--id=" + gatewaySenderId,
+        "--parallel=false",
+        "--remote-distributed-system-id=" + receiverSystemId);
+
+    String createSenderRegionCommand = String.join(GFSH_COMMAND_SEPARATOR,
+        "create region",
+        "--name=" + REGION_NAME,
+        "--type=" + RegionShortcut.REPLICATE.name(),
+        "--gateway-sender-id=" + gatewaySenderId);
+
+    gfshRule.execute(connectToSenderLocatorCommand, startGatewaySenderCommand);
+
+    // There is a bug in the GFSH create gateway-sender command where it returns before the system
+    // has recognized the creation status: GEODE-6777. We can remove the following when that bug is
+    // fixed.
+    try {
+      Thread.sleep(1000);
+    } catch (InterruptedException e) {
+      // Don't care
+    }
+
+    gfshRule.execute(connectToSenderLocatorCommand, createSenderRegionCommand);
+
+    String connectToReceiverLocatorCommand =
+        "connect --locator=localhost[" + receiverLocatorPort + "]";
+    String startGatewayReceiverCommand = "create gateway-receiver";
+    String createReceiverRegionCommand = String.join(GFSH_COMMAND_SEPARATOR,
+        "create region",
+        "--name=" + REGION_NAME,
+        "--type=" + RegionShortcut.REPLICATE.name());
+
+    gfshRule.execute(connectToReceiverLocatorCommand, startGatewayReceiverCommand,
+        createReceiverRegionCommand);
+
+    // Deploy function to members
+    String functionJarPath =
+        newJarForFunctionClass(GetEventsReceivedCountFunction.class, "function.jar");
+    String deployCommand = "deploy --jar=" + functionJarPath;
+    String listFunctionsCommand = "list functions";
+
+    gfshRule.execute(connectToReceiverLocatorCommand, deployCommand, listFunctionsCommand);
+  }
+
+  @After
+  public void stopClusters() {
+    String stopReceiverServerCommand = "stop server --dir=" + receiverServerFolder;
+    String stopSenderServerCommand = "stop server --dir=" + senderServerFolder;
+    String stopReceiverLocatorCommand = "stop locator --dir=" + receiverLocatorFolder;
+    String stopSenderLocatorCommand = "stop locator --dir=" + senderLocatorFolder;
+
+    gfshRule.execute(stopReceiverServerCommand, stopSenderServerCommand, stopReceiverLocatorCommand,
+        stopSenderLocatorCommand);
+  }
+
+  @Test
+  public void whenPerformingOperations_thenGatewayReceiverEventsReceivedIncreases() {
+    // TODO: perform put
```

Comment:
There's some TODOs, I think they are all covered and these are leftovers missed by a squash